task8/ProfitApp/src/main/java/com/bsu/fpmi/ProfitApp/AdsServlet.java

```diff
@@ -0,0 +1,86 @@
+package com.bsu.fpmi.ProfitApp;
+import com.google.gson.Gson;
+import javax.servlet.*;
+import javax.servlet.http.*;
+import javax.servlet.annotation.*;
+import java.io.IOException;
+
+@WebServlet(name = "adsServlet", value = "/ads")
+public class AdsServlet extends HttpServlet {
+    private AdCollection adCollection;
+
+    @Override
+    public void init() throws ServletException {
+        adCollection = new AdCollection();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String id = request.getParameter("id");
+        response.setContentType("json");
+        response.getWriter().println((new Gson()).toJson(adCollection.get(id)));
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String type = request.getParameter("type");
+        switch (type) {
+            case "search": {
+                int skip = Integer.parseInt(request.getParameter("skip"));
```

Comment:
You don't need to enclose `case` body in `{}`. It's enough to have `break` in the end